### PR TITLE
Deprecations, backport bug fixes.

### PR DESCRIPTION
**Pull request by JohnLCaron**
This change is [Reviewable](https://reviewable.io/reviews/unidata/netcdf-java/436)

### CODE REVIEW

**Comment by JohnLCaron** [comment]
Not sure these are real failures. Can you have a look?

File: cdm-test/src/test/java/ucar/nc2/dt/grid/TestReadProblem.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+import org.junit.Test;
+import ucar.unidata.util.test.TestDir;
+
+/** Open specific problem dataset with GridDataset.open(). */
+public class TestReadProblem {
+
+  @Test
+  public void doOne() throws Exception {
+    TestReadandCount.doOne(TestDir.cdmUnitTestDir + "formats/grib2/", "eta218.wmo", 57, 16, 20, 11);
```
Comment (lesserwhirls) [inline cdm-test/src/test/java/ucar/nc2/dt/grid/TestReadProblem.java:18]: I think this needs the `@Category(NeedsCdmUnitTest.class)` annotation.

**Comment by lesserwhirls** [comment]
I did a rebase of this over at Unidata/netcdf-java#442 (looks like it branched off of old master and not maint-5.x). Can you have a look over there to see if it looks ok to you?

**Review by lesserwhirls — COMMENTED** [review]
> Reviewed 18 of 29 files at r1, 11 of 11 files at r2.
> *[Reviewable](https://reviewable.io/reviews/unidata/netcdf-java/436)* status: all files reviewed, 1 unresolved discussion (waiting on @JohnLCaron)

**Comment by lesserwhirls** [comment]
Looks like you got it here. I'll close mine and go with this one. Thanks @JohnLCaron!

**Pull request merged**